lib/auth/oidc.go

```diff
@@ -265,6 +265,28 @@ type oidcAuthResponse struct {
 	claims jose.Claims
 }
 
+func checkEmailVerifiedClaim(claims jose.Claims) error {
+	claimName := "email_verified"
+	unverified := func() error { return trace.AccessDenied("email not verified by OIDC provider") }
+
+	emailVerified, hasEmailVerifiedClaim, _ := claims.StringClaim(claimName)
+	if hasEmailVerifiedClaim && emailVerified == "false" {
+		return unverified()
+	} else {
+		data, ok := claims[claimName]
+		if !ok {
+			return nil
+		}
+
+		emailVerified, ok := data.(bool)
+		if ok && !emailVerified {
```

Comment:
If `ok` is false because the type assertion fails, this check will succeed.

Shouldn't we return an error that we couldn't verify in this case?